src/node.cc

```diff
@@ -858,8 +859,12 @@ static ExitCode InitializeNodeWithArgsInternal(
     auto cwd = Environment::GetCwd(Environment::GetExecPath(*argv));
 
     for (const auto& file_path : file_paths) {
-      std::string path = cwd + kPathSeparator + file_path;
-      per_process::dotenv_file.ParsePath(path);
+      if (file_path.is_absolute()) {
+        per_process::dotenv_file.ParsePath(file_path.string());
+      } else {
+        std::string path = cwd + kPathSeparator + file_path.string();
+        per_process::dotenv_file.ParsePath(path);
+      }
```

Comment:
Could you add a comment explaining why it is necessary to construct an absolute path here?

Reply:
I don't see why any of this is necessary, so I've opened https://github.com/nodejs/node/pull/51425 as an alternative.